### PR TITLE
Menu: make link in item clickable by keyboard

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: js/ui/collection/ui.collection_widget.base.js [hunk 1]
```diff
@@ -86,7 +67,7 @@ const CollectionWidget = Widget.inherit({
         };
         return extend(this.callBase(), {
             space: space,
-            enter: enter,
+            enter: this._enterKeyHandler,
```
Comment (ksercs) [inline js/ui/collection/ui.collection_widget.base.js:70]: Seems like context is lost here. Why test does not show the mistake?

File: js/ui/context_menu/ui.menu_base.js [hunk 2]
```diff
@@ -165,7 +165,20 @@ class MenuBase extends HierarchicalCollectionWidget {
...
+                const $itemElement = $(this.option('focusedElement'));
+                if(!$itemElement.length) {
+                    return;
+                }
+                this._enterKeyHandler(e);
+
+                const itemData = this._getItemData($itemElement);
+                if(itemData.url) {
+                    const link = $itemElement.get(0).getElementsByClassName(DX_ITEM_URL_CLASS)[0];
+                    link && link.click();
```
Comment (ksercs) [inline js/ui/context_menu/ui.menu_base.js]:
```suggestion
                    link?.click();
```

File: testing/tests/DevExpress.ui.widgets/menuBase.tests.js [hunk 3]
```diff
@@ -1058,6 +1058,42 @@ QUnit.module('Keyboard navigation', () => {
 
         this.clock.restore();
     });
+
+    QUnit.test('Url in item should be clicked by enter button if item.url is set', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menuBase.tests.js]:
```suggestion
    QUnit.test('Url in item should be clicked by enter key press if item.url is set', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets/menuBase.tests.js [hunk 4]
```diff
@@ -1058,6 +1058,42 @@ QUnit.module('Keyboard navigation', () => {
...
+        menuItem.click = clickSpy;
+
+        keyboardMock(menuBase.element)
+            .keyDown('down')
+            .keyDown('enter');
+
+        assert.ok(clickSpy.calledOnce);
+    });
+
+    QUnit.test('Error should not be raised if item.url and item.template are set and item clicked by enter button', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menuBase.tests.js]:
```suggestion
    QUnit.test('Error should not be raised if item.url and item.template are set and enter is pressed', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets/menuBase.tests.js [hunk 5]
```diff
@@ -1058,6 +1058,42 @@ QUnit.module('Keyboard navigation', () => {
...
+            focusStateEnabled: true
+        });
+
+        try {
+            keyboardMock(menuBase.element)
+                .keyDown('down')
+                .keyDown('enter');
+            assert.ok(true, 'There is no error');
+        } catch(e) {
+            assert.ok(false, 'There is error');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menuBase.tests.js]:
```suggestion
            assert.ok(false, `Error is raised: ${e.message}`);
```

File: js/ui/collection/ui.collection_widget.base.js (hunk 1 above)
Comment (Krijovnick) [reply js/ui/collection/ui.collection_widget.base.js:70]: Context is not lost - https://github.com/DevExpress/DevExtreme/blob/1d6c8f5d2ea0ca80cad8db96bcf767262dc3686f/js/ui/widget/ui.widget.js#L330

**Review by ksercs — APPROVED** [review]

**Pull request merged**